types/encoding.go

```diff
@@ -0,0 +1,153 @@
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.SiacoinOutputs))))
+	w.Write(length)
+	for i := range t.SiacoinOutputs {
+		t.SiacoinOutputs[i].MarshalSia(w)
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.FileContracts))))
+	w.Write(length)
+	for i := range t.FileContracts {
+		enc.Encode(t.FileContracts[i])
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.FileContractRevisions))))
+	w.Write(length)
+	for i := range t.FileContractRevisions {
+		enc.Encode(t.FileContractRevisions[i])
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.StorageProofs))))
+	w.Write(length)
+	for i := range t.StorageProofs {
+		enc.Encode(t.StorageProofs[i])
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.SiafundInputs))))
+	w.Write(length)
+	for i := range t.SiafundInputs {
+		enc.Encode(t.SiafundInputs[i])
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.SiafundOutputs))))
+	w.Write(length)
+	for i := range t.SiafundOutputs {
+		t.SiafundOutputs[i].MarshalSia(w)
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.MinerFees))))
+	w.Write(length)
+	for i := range t.MinerFees {
+		t.MinerFees[i].MarshalSia(w)
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.ArbitraryData))))
+	w.Write(length)
+	for i := range t.ArbitraryData {
+		binary.LittleEndian.PutUint64(length, uint64(len((t.ArbitraryData[i]))))
+		w.Write(length)
+		w.Write(t.ArbitraryData[i])
+	}
+	binary.LittleEndian.PutUint64(length, uint64(len((t.TransactionSignatures))))
+	w.Write(length)
+	for i := range t.TransactionSignatures {
+		err := t.TransactionSignatures[i].MarshalSia(w)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// MarshalSia implements the encoding.SiaMarshaler interface.
+func (sci SiacoinInput) MarshalSia(w io.Writer) error {
+	w.Write(sci.ParentID[:])
+	return sci.UnlockConditions.MarshalSia(w)
+}
+
+// MarshalSia implements the encoding.SiaMarshaler interface.
+func (sco SiacoinOutput) MarshalSia(w io.Writer) error {
+	sco.Value.MarshalSia(w)
+	_, err := w.Write(sco.UnlockHash[:])
+	return err
+}
+
+// MarshalSia implements the encoding.SiaMarshaler interface.
+func (sfi SiafundInput) MarshalSia(w io.Writer) error {
+	w.Write(sfi.ParentID[:])
+	sfi.UnlockConditions.MarshalSia(w)
+	_, err := w.Write(sfi.ClaimUnlockHash[:])
+	return err
+}
+
+// MarshalSia implements the encoding.SiaMarshaler interface.
+func (sfo SiafundOutput) MarshalSia(w io.Writer) error {
+	sfo.Value.MarshalSia(w)
+	w.Write(sfo.UnlockHash[:])
+	return sfo.ClaimStart.MarshalSia(w)
+}
+
+// MarshalSia implements the encoding.SiaMarshaler interface.
+func (cf CoveredFields) MarshalSia(w io.Writer) error {
+	encoding.NewEncoder(w).Encode(cf.WholeTransaction)
+	fields := [][]uint64{
+		cf.SiacoinInputs,
+		cf.SiacoinOutputs,
+		cf.FileContracts,
+		cf.FileContractRevisions,
+		cf.StorageProofs,
+		cf.SiafundInputs,
+		cf.SiafundOutputs,
+		cf.MinerFees,
+		cf.ArbitraryData,
+		cf.TransactionSignatures,
+	}
```

Comment:
I'm pretty sure that if you messed this up (we should actually verify that by intentionally messing it up), the chain would be unable to sync, and you'd have a node that was stuck at a very low block height.